test/distributed/org/apache/cassandra/distributed/mock/nodetool/NetstatsTest.java

```diff
@@ -0,0 +1,599 @@
+                           '}';
+                }
+            }
+        }
+    }
+
+    private static final class NetstatResults
+    {
+        private final List<NodeToolResultWithOutput> netstatOutputs = new ArrayList<>();
+
+        public void add(NodeToolResultWithOutput result)
+        {
+            netstatOutputs.add(result);
+        }
+
+        public void assertSuccessful()
+        {
+            for (final NodeToolResultWithOutput result : netstatOutputs)
+            {
+                Assert.assertEquals(result.getResult().getRc(), 0);
+                Assert.assertTrue(result.getStderr().isEmpty());
+            }
+        }
+    }
+
+    private static class NetstatsCallable implements Callable<NetstatResults>
+    {
+        private final IInvokableInstance node;
+
+        public NetstatsCallable(final IInvokableInstance node)
+        {
+            this.node = node;
+        }
+
+        public NetstatResults call() throws Exception
+        {
+            final NetstatResults results = new NetstatResults();
+
+            boolean sawAnyStreamingOutput = false;
+
+            while (true)
+            {
+                final NodeToolResultWithOutput result = NodetoolUtils.nodetool(node, false, "netstats");
+
+                System.out.println(result.getStdout());
+
+                if (!sawAnyStreamingOutput)
+                {
+                    if (result.getStdout().contains("Receiving") || result.getStdout().contains("Sending"))
+                    {
+                        sawAnyStreamingOutput = true;
+                    }
+                }
+
+                if (sawAnyStreamingOutput && (!result.getStdout().contains("Receiving") && !result.getStdout().contains("Sending")))
+                {
+                    break;
+                }
+
+                results.add(result);
+
+                Thread.currentThread().sleep(1000);
+            }
+
+            return results;
+        }
+    }
+
+    private void changeReplicationFactor()
+    {
+        try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+             Session s = c.connect())
+        {
+            s.execute("ALTER KEYSPACE netstats_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 2};");
+        }
+    }
+
+    private void createTable(Cluster cluster)
+    {
+        // replication factor is 1
+        cluster.schemaChange("CREATE KEYSPACE netstats_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+        cluster.schemaChange("CREATE TABLE netstats_test.test_table (id uuid primary key);");
+    }
+
+    private void populateData()
+    {
+        try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+             Session s = c.connect("netstats_test"))
+        {
+
+            for (int i = 0; i < 500000; i++)
```

Comment:
this causes periodic timeouts as the test takes 5m30s and the timeout for the class is 6m.  in testing I find that `100_000` (100k) is around 3m30s to 4m so should be safe in CI.